api/server/server.go

```diff
@@ -43,15 +43,22 @@ type Server struct {
 	Datastore    models.Datastore
 }
 
-func New(ctx context.Context, ds models.Datastore, mq models.MessageQueue, r *runner.Runner, tasks chan task.Request, enqueue models.Enqueue) *Server {
+type Components struct {
```

Comment:
> if we have an API that embeds many other components (current situation, can probably get worse), the user creating its own server will have provide each other component to the API Server in its instantiation.

Aren't what abstractions for? 

> But still the server will be the main component on this server and it must have access to other replaceable components.

It seems that again I failed to make my point. If you have the design I proposed, the only components it will need is a channel of enqueueing events (bi-directional, for the sake of async) and datastore. No need to peg the runner, the mq or anything. Namely because all of these will be running separately somewhere else - serving `server` through this channel.

And, if the initialization is the problem, perhaps the solution is to create an abstraction to help anyone customizing IronFunctions.